pkg/cloud/services/awsnode/cni.go

```diff
@@ -272,22 +271,23 @@ func (s *Service) deleteResource(ctx context.Context, remoteClient client.Client
 			return fmt.Errorf("deleting resource %s: %w", key, err)
 		}
 		s.scope.Debug(fmt.Sprintf("resource %s was not found, no action", key))
-	} else {
-		// resource found, delete if no label or not managed by helm
-		if val, ok := obj.GetLabels()[konfig.ManagedbyLabelKey]; !ok || val != "Helm" {
-			if err := remoteClient.Delete(ctx, obj, &client.DeleteOptions{}); err != nil {
-				if !apierrors.IsNotFound(err) {
-					return fmt.Errorf("deleting %s: %w", key, err)
-				}
-				s.scope.Debug(fmt.Sprintf(
-					"resource %s was not found, not deleted", key))
-			} else {
-				s.scope.Debug(fmt.Sprintf("resource %s was deleted", key))
-			}
-		} else {
-			s.scope.Debug(fmt.Sprintf("resource %s is managed by helm, not deleted", key))
+		return nil
+	}
+	// Check if the "prevent-deletion" label is present and set to "true", skip if present
+	labels := obj.GetLabels()
+	if value, exists := labels["prevent-deletion"]; exists && value == "true" {
```

Comment:
what do you think about checking the key presence instead of comparing it to `true` value? It's easy to make a mistake for the user when setting it because keys and values are just strings and we can't validate them. Adding "prevent-deletion" even with an empty value can be enough.

Reply:
Thanks for the review. That sounds fine to me.